### PR TITLE
search: fix highlight matches count

### LINKED ISSUE #21443: matches report lines instead of highlights
Looking at the screenshot, on the right side it says "275 matches, 265 more" which 
implies we see 10 matches, however counting the highlights I see 12 matches for the first file.
We seem to be reporting lines and not highlights.

https://sourcegraph.com/search?q=context:global+repo:github.com/sourcegraph/sourcegraph+resolver+lang:go&patternType=literal

Screenshot 2021-05-27 at 13 02 40

### PR DESCRIPTION
**Pull request by novoselrok**
Fixes #21443

Current:

Screenshot 2021-08-23 at 11 11 51

Fixed (588 - 576 = 12, which corresponds to the number of visible highlight ranges):

Screenshot 2021-08-23 at 11 12 24

### CODE REVIEW

File: client/shared/src/components/FileMatch.tsx [hunk 1]
```diff
@@ -154,22 +176,14 @@ export const FileMatch: React.FunctionComponent<Props> = props => {
             repoStars: result.repoStars,
         }
     } else {
-        const length = items.length - subsetMatches
+        const length = highlightRangesCount - collapsedHighlightRangesCount
```
Comment (novoselrok) [inline client/shared/src/components/FileMatch.tsx:179]: This was the line that incorrectly calculated the remaining matches.

**Review by camdencheek — APPROVED** [review]
Nice! Looking forward to having this fixed 😄

**Comment by rvantonder** [comment]
changelog pls 🤗  (for our own tracking purposes)

File: client/shared/src/components/FileMatch.tsx [hunk 2]
```diff
@@ -70,28 +71,41 @@ interface Props extends SettingsCascadeProps {
...
+
 export const FileMatch: React.FunctionComponent<Props> = props => {
-    const [subsetMatches, setSubsetMatches] = useState(10)
-    useEffect(() => {
-        const subsetMatches = parseInt(localStorage.getItem(SUBSET_COUNT_KEY) || '', 10)
-        if (!isNaN(subsetMatches)) {
-            setSubsetMatches(subsetMatches)
+    // The number of lines of context to show before and after each match.
+    const context = useMemo(() => {
+        if (props.location.pathname === '/search') {
```
Comment (limitedmage) [inline client/shared/src/components/FileMatch.tsx:79]: This means this won't affect other ways in which results are displayed, such as the code intel panels at the bottom of file views, right?
This seems unintuitive. I wonder if there is a better way to do this, for example, injection the number of context lines as a prop instead.

Comment (limitedmage) [reply client/shared/src/components/FileMatch.tsx:79]: I also wonder if this should be added to the changelog? Is this an actual change in current functionality? It seems unrelated to the issue.

**Review by limitedmage — APPROVED** [review]

File: client/shared/src/components/FileMatch.tsx (hunk 2 above)
Comment (novoselrok) [reply client/shared/src/components/FileMatch.tsx:79]: This was copied over from `FileMatchChildren.tsx` when I moved the match grouping functionality one level up. I agree that it's a bit weird to guard this with a specific pathname. I'll ask in Slack if anyone knows why this is the case.

**Pull request merged**